### PR TITLE
Update SDK README.md

### PR DESCRIPTION
**Pull request by ckadner**
Add a paragraph to SDK `README.md` describing the classes/functions that have been extended from the KFP SDK. 

---

### Extensions to the Kubeflow Pipelines SDK

In addition to the functionality provided by the Kubeflow Pipelines
[SDK](https://www.kubeflow.org/docs/pipelines/sdk/sdk-overview/) the `kfp-tekton`
SDK provides a `TektonCompiler` and a `TektonClient`:

`TektonCompiler`:

 - `kfp_tekton.compiler.TektonCompiler.compile` compiles Python DSL code into a 
   YAML file containing a Tekton `PipelineRun` which can be deployed directly to
   a Tekton enabled Kubernetes cluster or uploaded to the Kubeflow Pipelines
   dashboard with the Tekton backend.
   
`TektonClient`:

 - `kfp_tekton.TektonClient.create_run_from_pipeline_func` compiles DSL pipeline
   function and runs the pipeline on a Kubernetes cluster with KFP and Tekton

---

/cc @Tomcli

### CODE REVIEW

**Comment by ckadner** [comment]
Since @Tomcli also wrote a new README in #211 I only added a brief description of the two main classes in the KFP SDK which we are extending and what they are used for. I suggest to add Tommy's new user doc under the `sdk/` directory instead of the `samples/` folder and link it in the `sdk/README.md` here once PR #211 is merged.

File: sdk/README.md [hunk 1]
```diff
@@ -1,31 +1,51 @@
...
 
-The Kubeflow Pipelines [SDK](https://www.kubeflow.org/docs/pipelines/sdk/sdk-overview/) 
+The Kubeflow Pipelines [SDK](https://www.kubeflow.org/docs/pipelines/sdk/sdk-overview/)
 allows data scientists to define end-to-end machine learning and data pipelines.
 The output of the Kubeflow Pipelines SDK compiler is YAML for [Argo](https://github.com/argoproj/argo).
-We are extending the compiler of the Kubeflow Pipelines SDK to generate YAML
-for [Tekton](https://github.com/tektoncd/pipeline).
+
+The `kfp-tekton` SDK is extending the `Compiler` and the `Client` of the Kubeflow
+Pipelines SDK to generate YAML for [Tekton](https://github.com/tektoncd/pipeline).
```
Comment (animeshsingh) [inline sdk/README.md:8]:
```suggestion
Pipelines SDK to generate YAML for [Tekton](https://github.com/tektoncd/pipeline), and consequently upload and launch them with Kubeflow Pipelines engine backed by Tekton
```

**Comment by ckadner** [comment]
@animeshsingh @Tomcli -- this is what would show up on PyPi once we draw a release 

https://test.pypi.org/project/kfp-tekton/0.2.0rc3/

**Pull request merged**